universal-login-react/src/ui/Transfer/Transfer.tsx

```diff
@@ -24,27 +24,20 @@ interface ErrorsProps {
   recipientError: boolean;
 }
 
-export const Transfer = ({deployedWallet, transferDetails, updateTransferDetailsWith, tokenDetails, onSendClick, transferClassName}: TransferProps) => {
+export const Transfer = ({deployedWallet, transferDetails, updateTransferDetailsWith, tokenDetailsWithBalance, tokenDetails, onSendClick, transferClassName}: TransferProps) => {
   const [errors, setErrors] = useState<ErrorsProps>({amountError: false, recipientError: false});
-  const [tokenDetailsWithBalance, setTokenDetailsWithBalance] = useState<TokenDetailsWithBalance[]>([]);
 
-  const {sdk, contractAddress} = deployedWallet;
-  useAsyncEffect(() => sdk.subscribeToBalances(contractAddress, setTokenDetailsWithBalance), []);
-  const balance = getBalanceOf(tokenDetails.symbol, tokenDetailsWithBalance);
-
-  const {amount, to, gasParameters} = transferDetails;
-
-  const onClick = () => {
-    const errorsCalculated = {
-      amountError: !isValidAmount(balance, amount),
-      recipientError: !isValidRecipient(to),
-    };
-    setErrors(errorsCalculated);
-    areInputsValid(errorsCalculated) && onSendClick();
+  const onTransferClick = async () => {
+    try {
+      await onSendClick();
+    } catch (error) {
+      setErrors({
+        amountError: error.errorType !== 'InvalidAddressOrEnsName' && true,
+        recipientError: error.errorType !== 'InvalidAmount' && true,
+      });
+    }
```

Comment:
We can now move ErrorProps to app layer and use them as a state instead as props.
That way we can have:
```ts
const {errors} = await transferService.transfer(transferDetails);
if (errors) {
  setErrors(errors);
} else {
  modalService.showModal('transfer');
}
```

or even better:
```ts
  history.push('/transfer')
```

Reply:
I planned change whole transfer flow into /transfer route and flow like in BackupCodes after this pr